src/router_core/modules/stuck_delivery_detection/delivery_tracker.c

```diff
@@ -46,15 +46,19 @@ struct tracker_t {
 
 static void check_delivery_CT(qdr_core_t *core, qdr_link_t *link, qdr_delivery_t *dlv)
 {
-    if (!dlv->stuck && ((core->uptime_ticks - link->core_ticks) > stuck_age)) {
-        dlv->stuck = true;
-        link->deliveries_stuck++;
-        core->deliveries_stuck++;
-        if (link->deliveries_stuck == 1)
-            qd_log(core->log, QD_LOG_INFO,
-                   "[C%"PRIu64"][L%"PRIu64"] "
-                   "Stuck delivery: At least one delivery on this link has been undelivered/unsettled for more than %d seconds",
-                   link->conn ? link->conn->identity : 0, link->identity, stuck_age);
+    // DISPATCH-2036: ignore "infinitely long" streaming messages (like TCP
+    // adaptor deliveries)
+    if (dlv->msg && !qd_message_is_streaming(dlv->msg)) {
+        if (!dlv->stuck && ((core->uptime_ticks - link->core_ticks) > stuck_age)) {
```

Comment:
```suggestion
    // DISPATCH-2036: ignore "infinitely long" streaming messages (like TCP
    // adaptor deliveries)
    if (dlv->msg && qd_message_is_streaming(dlv->msg)) {
        return;
    }
    if (!dlv->stuck && ((core->uptime_ticks - link->core_ticks) > stuck_age)) {
        [....]
```

Mostly aesthetics, but keeping the level of indentation down is often helpful...